packages/editor-ui/src/store.ts

```diff
@@ -777,7 +782,13 @@ export const store = new Vuex.Store({
 		allNodes: (state): INodeUi[] => {
 			return state.workflow.nodes;
 		},
-		nodeByName: (state) => (nodeName: string): INodeUi | null => {
+		nodesByName: (state: IRootState): {[name: string]: INodeUi} => {
+			return state.workflow.nodes.reduce((accu: {[name: string]: INodeUi}, node) => {
+				accu[node.name] = node;
+				return accu;
+			}, {});
+		},
+		getNodeByName: (state) => (nodeName: string): INodeUi | null => {
```

Comment:
can't we use our getter `nodesByName` for this instead of looping? Should look like this:

```
getNodeByName: (state, getters) => (nodeName: string): INodeUi | null => {
  return getters.nodesByName[nodeName] || null;
}
```

nit-pick: don't know if we depend on it but returning `undefined` instead of `null` feels more intuivite to me

Reply:
I was planning to replace it with `nodesByName` but was scared to do so without sufficient testing because it would change Vue reactivity.. your suggestion is great because it does not affect reactivity. thanks